modules/builders/src/prerender/index.ts

```diff
@@ -103,93 +102,75 @@ async function _renderUniversal(
-              outputPath,
-              browserOptions.deployUrl || '',
-              normalizedStylesOptimization.inlineCritical ? 'true' : 'false',
-              normalizedStylesOptimization.minify ? 'true' : 'false',
-              ...routesShard,
-            ])
-              .on('message', (data) => {
-                if (data.success === false) {
-                  reject(
-                    new Error(`Unable to render ${data.outputIndexPath}.\nError: ${data.error}`),
-                  );
-
-                  return;
-                }
-
-                if (data.logLevel) {
-                  spinner.stop();
-                  context.logger.log(data.logLevel, data.message);
-                  spinner.start();
-                }
-              })
-              .on('exit', resolve)
-              .on('error', reject);
-          }),
-      );
-
-      await Promise.all(childProcesses);
-    } catch (error) {
-      spinner.fail(`Prerendering routes to ${outputPath} failed.`);
-
-      return { success: false, error: error.message };
-    }
-
-    spinner.succeed(`Prerendering routes to ${outputPath} complete.`);
+  const { baseOutputPath = '' } = serverResult;
+
+  const workerArgs: WorkerSetupArgs = {
+    indexFile,
+    deployUrl: browserOptions.deployUrl || '',
+    inlineCriticalCss: !!normalizedStylesOptimization.inlineCritical,
+    minifyCss: !!normalizedStylesOptimization.minify,
+  };
+  const worker = new JestWorker(path.join(__dirname, 'worker.js'), {
+    exposedMethods: ['render'],
+    enableWorkerThreads: true,
+    numWorkers: numProcesses,
+    setupArgs: [workerArgs],
+  });
+  try {
+    worker.getStdout().pipe(process.stdout);
+    worker.getStderr().pipe(process.stderr);
+
+    // We need to render the routes for each locale from the browser output.
+    for (const outputPath of browserResult.outputPaths) {
+      const localeDirectory = path.relative(browserResult.baseOutputPath, outputPath);
+      const serverBundlePath = path.join(baseOutputPath, localeDirectory, 'main.js');
+      if (!fs.existsSync(serverBundlePath)) {
+        throw new Error(`Could not find the main bundle: ${serverBundlePath}`);
+      }
 
-    if (browserOptions.serviceWorker) {
-      spinner.start('Generating service worker...');
+      const spinner = ora(`Prerendering ${routes.length} route(s) to ${outputPath}...`).start();
       try {
-        await augmentAppWithServiceWorker(
-          root,
-          projectRoot,
-          normalize(outputPath),
-          browserOptions.baseHref || '/',
-          browserOptions.ngswConfigPath,
-        );
+        const results = (await Promise.all(
+          routes.map((route) => (worker as any).render(outputPath, serverBundlePath, route)),
+        )) as RenderResult[];
+        let numErrors = 0;
+        for (const { errors, warnings } of results) {
+          errors?.forEach((e) => context.logger.error(e));
+          warnings?.forEach((e) => context.logger.warn(e));
```

Comment:
Before printing anything, when the spinner is running you need to stop and start it, as otherwise it will overlap the  current spinner and cause broken output.


```suggestion
          spinner.stop();
          errors?.forEach((e) => context.logger.error(e));
          warnings?.forEach((e) => context.logger.warn(e));
           spinner.start();
```

Reply:
Bump!

Reply:
Yep, it's on my list; plan to work on it this weekend.